### PR TITLE
feat(model): add Qwen3.8-27B validation and cookbook

### PR DESCRIPTION
**Pull request by spwahaha**
## Motivation

Validate and document text-only inference support for `Qwen/Qwen3.8-27B`.

Qwen3.8-27B declares the existing Qwen3.5 dense architecture, so it can reuse the current SGL-JAX implementation without introducing a new runtime model class.

## Modifications

- Register `Qwen/Qwen3.8-27B` in the test utilities.
- Add a Qwen3.8 TPU integration test that reuses the existing Qwen3.5 dense launch configuration.
- Add a validated Qwen3.8-27B cookbook covering:
  - BF16 deployment on TPU v6e-4 with TP4.
  - Hybrid recurrent-state cache requirements.
  - Non-thinking chat completion.
  - Streamed reasoning.
  - Structured tool calling.
  - GSM8K and MMLU accuracy results.
  - Serving benchmark results.
- Add Qwen3.8-27B to the cookbook navigation and hardware indexes.

## Accuracy Tests

Test environment:

- Model: `Qwen/Qwen3.8-27B`
- Hardware: TPU v6e-4, single host
- Tensor parallelism: 4
- Precision: BF16

### MMLU integration test

- Examples: 100
- Accuracy: `0.900`
- Required threshold: `> 0.70`

### GSM8K

```bash
evalscope eval \
  --model Qwen/Qwen3.8-27B \
  --api-url http://127.0.0.1:30000/v1/chat/completions \
  --api-key EMPTY \
  --eval-type openai_api \
  --datasets gsm8k \
  --eval-batch-size 8 \
  --limit 200 \
  --generation-config '{"chat_template_kwargs":{"enable_thinking":true},"temperature":1.0,"top_p":0.95,"top_k":20,"min_p":0.0,"presence_penalty":0.0,"repetition_penalty":1.0,"max_tokens":32768}'
```

Result:

| Dataset | Examples | Accuracy |
|---|---:|---:|
| GSM8K | 200 | 97.5% |

The OpenAI-compatible API was also manually validated for non-thinking chat, streamed reasoning, and structured tool calling.

## Benchmarking and Profiling

```bash
python -m sgl_jax.bench_serving \
  --backend sgl-jax \
  --dataset-name random \
  --num-prompts 100 \
  --random-input 512 \
  --random-output 128 \
  --max-concurrency 8 \
  --random-range-ratio 1 \
  --warmup-requests 0 \
  --tokenizer Qwen/Qwen3.8-27B
```

```text
============ Serving Benchmark Result ============
Backend:                                 sgl-jax
Traffic request rate:                    inf
Max request concurrency:                 8
Successful requests:                     100
Benchmark duration (s):                  32.67
Total input tokens:                      51200
Total input text tokens:                 51200
Total generated tokens:                  12800
Total generated tokens (retokenized):    12682
Total cached tokens:                     0
Cache hit rate:                          0.0000
Request throughput (req/s):              3.06
Input token throughput (tok/s):          1567.07
Output token throughput (tok/s):         391.77
Peak output token throughput (tok/s):    456.00
Peak concurrent requests:                16
Total token throughput (tok/s):          1958.84
Concurrency:                             7.72
----------------End-to-End Latency----------------
Mean E2E Latency (ms):                   2521.53
Median E2E Latency (ms):                 2531.10
P90 E2E Latency (ms):                    2539.32
P99 E2E Latency (ms):                    2543.01
---------------Time to First Token----------------
Mean TTFT (ms):                          169.93
Median TTFT (ms):                        153.21
P99 TTFT (ms):                           303.33
-----Time per Output Token (excl. 1st token)------
Mean TPOT (ms):                          18.52
Median TPOT (ms):                        18.47
P99 TPOT (ms):                           19.67
---------------Inter-Token Latency----------------
Mean ITL (ms):                           18.52
Median ITL (ms):                         17.53
P95 ITL (ms):                            17.86
P99 ITL (ms):                            18.52
Max ITL (ms):                            280.76
==================================================
```

This is a fixed-shape synthetic workload with 512 input tokens, 128 requested output tokens, concurrency 8, and no benchmark warm-up requests.

## Checklist

- [x] Please use English, otherwise it will be closed.
- [x] The purpose of the PR, or link existing issues this PR will resolve.
- [x] The test plan, such as providing test command.
- [x] (Optional) The necessary documentation update.

### CODE REVIEW

File: docs/cookbook/autoregressive/Qwen/Qwen3.8.md [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+
+**Current Scope**: This recipe covers BF16 text generation only. Image and video input, MTP speculative decoding, FP8, context extension beyond 262,144 tokens, and Qwen3.8-2.4T-A95B are not currently validated in SGL-JAX.
+
+**License**: see the [Hugging Face model card](https://huggingface.co/Qwen/Qwen3.8-27B) for the authoritative license terms.
+
+## 2. Deployment
+
+### 2.1 Hardware Matrix
+
+| Model | TPU | Topology | Chips | `--tp-size` | Precision | Notes |
```
Comment (cjx0709) [inline docs/cookbook/autoregressive/Qwen/Qwen3.8.md:35]: Other dense-model cookbooks (Qwen3, Llama3.1, Gemma2, MiMo-7B, etc.) don't have a `Precision` column in this table — they all use `| Model | TPU | Topology | Chips | \`--tp-size\` | Notes |`. Suggest dropping the `Precision` column and folding BF16 into Notes, e.g.:

```
| Model | TPU | Topology | Chips | `--tp-size` | Notes |
|---|---|---|---|---|---|
| Qwen3.8-27B | **v6e-4** | 2x2 | 4 | 4 | BF16 text-only; single host. |
```

File: docs/cookbook/autoregressive/Qwen/Qwen3.8.md [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+  --eval-batch-size 8 \
+  --limit 200 \
+  --generation-config '{"chat_template_kwargs":{"enable_thinking":true},"temperature":1.0,"top_p":0.95,"top_k":20,"min_p":0.0,"presence_penalty":0.0,"repetition_penalty":1.0,"max_tokens":32768}'
+```
+
+**Test Results**
+
+| Model | Dataset | Metric | Subset | Num | Score | Date |
+|:---|:---|:---|:---|---:|---:|:---|
+| Qwen3.8-27B | GSM8K | Accuracy | main | 200 | **97.5%** | 2026-09-15 |
```
Comment (cjx0709) [inline docs/cookbook/autoregressive/Qwen/Qwen3.8.md:279]: A few format mismatches vs the other cookbooks (Qwen3, DeepSeek-R1, MiMo-7B, etc.):
- Dataset should be lowercase `gsm8k` (not `GSM8K`)
- Metric should be `AverageAccuracy` (not `Accuracy`)
- Score should be a plain decimal `0.975` without bold (not `**97.5%**`)
- Drop the `Date` column — no other cookbook has it

Suggested:
```
| Model | Dataset | Metric | Subset | Num | Score |
|:---|:---|:---|:---|---:|---:|
| Qwen3.8-27B | gsm8k | AverageAccuracy | main | 200 | 0.975 |
```

File: docs/cookbook/autoregressive/Qwen/Qwen3.8.md [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+EvalScope reported an average latency of 11.295 seconds and average output
+throughput of 54.13 tokens/s across these evaluation requests. Because the run
+used `--limit 200`, treat this as a sampled integration result rather than a
+formal full-dataset evaluation.
+
+### 4.2 Additional Validation — MMLU
+
+The initial integration run used the repository's 100-example MMLU evaluator. This result remains useful as loader and generation evidence, but it is separate from the canonical cookbook GSM8K benchmark.
+
+| Model | Dataset | Metric | Subset | Num | Score | Date |
```
Comment (cjx0709) [inline docs/cookbook/autoregressive/Qwen/Qwen3.8.md:290]: Same here — drop the `Date` column to stay consistent with the rest of the cookbooks.

**Review by cjx0709 — COMMENTED** [review]
Nice cookbook — content is solid and well-structured. A few formatting items to align with the existing cookbooks before merging.

One thing not covered by inline comments: the **parser key reference table** at the bottom of `docs/cookbook/autoregressive/index.md` should also be updated — the `qwen3_coder` row currently says "Qwen3-Coder variants" in the "Where used" column; please add a link to this Qwen3.8 recipe there as well.

File: docs/cookbook/autoregressive/Qwen/Qwen3.8.md (hunk 1 above)
Comment (spwahaha) [reply docs/cookbook/autoregressive/Qwen/Qwen3.8.md]: Addressed.

File: docs/cookbook/autoregressive/Qwen/Qwen3.8.md (hunk 2 above)
Comment (spwahaha) [reply docs/cookbook/autoregressive/Qwen/Qwen3.8.md]: Addressed.

File: docs/cookbook/autoregressive/Qwen/Qwen3.8.md (hunk 3 above)
Comment (spwahaha) [reply docs/cookbook/autoregressive/Qwen/Qwen3.8.md]: Addressed.

**Review by cjx0709 — APPROVED** [review]
LGTM
